src/mod_roster_rdbms.erl

```diff
@@ -275,40 +246,86 @@ read_subscription_and_groups(LUser, LServer, LJID, GSFunc, GRFunc) ->
-    SJID =
-    mongoose_rdbms:escape_string(jid:to_binary(jid:to_lower(JID))),
-    Nick = mongoose_rdbms:escape_string(Name),
-    SSubscription = mongoose_rdbms:escape_string(case Subscription of
-                        both -> <<"B">>;
-                        to -> <<"T">>;
-                        from -> <<"F">>;
-                        none -> <<"N">>
-                    end),
-    SAsk = mongoose_rdbms:escape_string(case Ask of
-               subscribe -> <<"S">>;
-               unsubscribe -> <<"U">>;
-               both -> <<"B">>;
-               out -> <<"O">>;
-               in -> <<"I">>;
-               none -> <<"N">>
-           end),
-    SAskMessage = mongoose_rdbms:escape_string(AskMessage),
-    [Username, SJID, Nick, SSubscription, SAsk, SAskMessage,
-     mongoose_rdbms:escape_string(<<"N">>),
-     mongoose_rdbms:escape_string(<<"">>),
-     mongoose_rdbms:escape_string(<<"item">>)].
-
-groups_to_string(#roster{us = {User, _Server},
-                         jid = JID, groups = Groups}) ->
-    Username = mongoose_rdbms:escape_string(User),
-    SJID = mongoose_rdbms:escape_string(jid:to_binary(jid:to_lower(JID))),
-    lists:foldl(fun (<<"">>, Acc) -> Acc;
-                    (Group, Acc) ->
-                        G = mongoose_rdbms:escape_string(Group),
-                        [[Username, SJID, G] | Acc]
-                end,
-                [], Groups).
+%% PgSQL returns CHAR as an integer
+%% MySQL returns CHAR as a string
+decode_subscription(<<X>>) -> decode_subscription(X);
+decode_subscription($B) -> both;
+decode_subscription($T) -> to;
+decode_subscription($F) -> from;
+decode_subscription($N) -> none.
+
+encode_subscription(both) -> <<"B">>;
+encode_subscription(to)   -> <<"T">>;
+encode_subscription(from) -> <<"F">>;
+encode_subscription(none) -> <<"N">>.
+
+decode_ask(<<X>>) -> decode_ask(X);
+decode_ask($S) -> subscribe;
+decode_ask($U) -> unsubscribe;
+decode_ask($B) -> both;
+decode_ask($O) -> out;
+decode_ask($I) -> in;
+decode_ask($N) -> none.
+
+encode_ask(subscribe) -> <<"S">>;
+encode_ask(unsubscribe) -> <<"U">>;
+encode_ask(both) -> <<"B">>;
+encode_ask(out)  -> <<"O">>;
+encode_ask(in)   -> <<"I">>;
+encode_ask(none) -> <<"N">>.
+
+record_to_row(#roster{us = {LUser, _Server},
+                      jid = JID, name = Nick, subscription = Subscription,
+                      ask = Ask, askmessage = AskMessage}) ->
+    BinJID = jid:to_binary(jid:to_lower(JID)),
+    ExtSubscription = encode_subscription(Subscription),
+    ExtAsk = encode_ask(Ask),
+    [LUser, BinJID, Nick, ExtSubscription, ExtAsk, AskMessage,
+     <<"N">>, <<>>, <<"item">>].
+
+groups_to_rows(#roster{us = {LUser, _LServer}, jid = JID, groups = Groups}) ->
+    BinJID = jid:to_binary(jid:to_lower(JID)),
+    lists:foldl(fun (<<>>, Acc) -> Acc;
+                    (Group, Acc) -> [[LUser, BinJID, Group] | Acc]
+                end, [], Groups).
+
+raw_to_record(LServer,
```

Comment:
we have `record_to_row` but `raw_to_record`. This is asking for typos. Please make it consistent.